pkg/config.go

```diff
@@ -25,34 +26,45 @@ func ParseConfig(kommonsClient *kommons.Client, configFile string) ([]logs.Searc
 	}
 
 	for _, backend := range searchConfig.Backends {
-		if backend.Kubernetes != nil {
-			client, err := k8s.GetKubeClient(kommonsClient, backend.Kubernetes)
-			if err != nil {
-				return nil, err
-			}
-			backend.Backend = &k8s.KubernetesSearch{
-				Client: client,
-			}
-			backends = append(backends, backend)
+		if err := AttachSearchAPIToBackend(&backend); err != nil {
+			logger.Errorf("Error attaching search api backend: %v", err)
 		}
+	}
+
+	return backends, nil
+}
+
+func AttachSearchAPIToBackend(backend *logs.SearchBackend) error {
+	kommonsClient, err := kommons.NewClientFromDefaults(logger.GetZapLogger())
+	if err != nil {
+		return err
+	}
 
-		if len(backend.Files) != 0 {
-			// If the paths are not absolute,
-			// They should be parsed with respect to the provided config file
-			for i, f := range backend.Files {
-				for j, p := range f.Paths {
-					if !filepath.IsAbs(p) {
-						backend.Files[i].Paths[j] = filepath.Join(filepath.Dir(configFile), p)
-					}
+	if backend.Kubernetes != nil {
+		client, err := k8s.GetKubeClient(kommonsClient, backend.Kubernetes)
+		if err != nil {
+			return err
+		}
+		backend.API = &k8s.KubernetesSearch{
+			Client: client,
+		}
+	}
+
+	if len(backend.Files) > 0 {
+		// If the paths are not absolute,
+		// They should be parsed with respect to the current path
+		for i, f := range backend.Files {
+			for j, p := range f.Paths {
+				if !filepath.IsAbs(p) {
+					currentPath, _ := os.Getwd()
+					backend.Files[i].Paths[j] = filepath.Join(currentPath, p)
```

Comment:
I think relative paths should be treated with respect to the config file.

You could potentially run the `apm-hub` binary from any directory whereas the config path and the file path referred to in the config are static.

Reply:
> I think relative paths should be treated with respect to the config file.
I disagree, 2 reasons:

1. The primary use-case being that it will run on kubernetes, the config file will be mounted as a ConfigMap where relative paths make no sense
2. If we somehow run this in a non k8s environment, keeping your config files where your logfiles are is unnatural

I think we should either:
- Not support relative paths at all (or)
- Take relative paths as a command line arg

Reply:
We have used relative to the config file in karina - but I think that was a specialized case as we were importing "base" configs into a final config. 

I think we do need to support the relative path for a testing perspective - It might also be needed if it is installed as part of an OEM package to expose log files for search - but then relative to working directory makes the most sense  The binary could be in program files, with a service that uses a different working directory e.g. a data or logs folder.